includes/embeds/class-amp-reddit-embed-handler.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Class AMP_Reddit_Embed_Handler
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Reddit_Embed_Handler
+ */
+class AMP_Reddit_Embed_Handler extends AMP_Base_Embed_Handler {
+	/**
+	 * Regex matched to produce output amp-reddit.
+	 *
+	 * @const string
+	 */
+	const URL_PATTERN = '#https?://(www\.)?reddit\.com/r/[^/]+/comments/.*#i';
+
+	/**
+	 * Script slug.
+	 *
+	 * @var string
+	 */
+	private static $script_src = 'https://cdn.ampproject.org/v0/amp-reddit-0.1.js';
+
+	/**
+	 * Register embed.
+	 */
+	public function register_embed() {
+		wp_embed_register_handler( 'amp-reddit', self::URL_PATTERN, array( $this, 'oembed' ) );
+	}
+
+	/**
+	 * Unregister embed.
+	 */
+	public function unregister_embed() {
+		wp_embed_unregister_handler( 'amp-reddit' );
+	}
+
+	/**
+	 * Include the required AMP Reddit scripts.
+	 */
+	public function get_scripts() {
```

Comment:
It should be unnecessary to implement `get_scripts()` because now the whitelist sanitizer will automatically discover the `amp-reddit` script as being required through sanitization when it runs across the `amp-reddit` element.